src/extension/features/budget/cover-overspending-from-future/index.js

```diff
@@ -0,0 +1,163 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { /* getSelectedMonth, */ getEntityManager, isCurrentRouteBudgetPage } from 'toolkit/extension/utils/ynab';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { l10n } from 'toolkit/extension/utils/toolkit';
+import { controllerLookup } from 'toolkit/extension/utils/ember';
+
+const BUTTON_CLASS = 'tk-button-cover-from-future';
+
+export class CoverOverspendingFromFuture extends Feature {
+  injectCSS() {
+    return require('./index.css');
+  }
+
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage();
+  }
+
+  invoke() {
+    this.verifyButtonDisplay();
+  }
+
+  observe(changedNodes) {
+    if (!this.shouldInvoke()) return;
+    if (changedNodes.has('budget-table-row is-sub-category is-checked')) {
+      this.invoke();
+    }
+  }
+
+  verifyButtonDisplay() {
+    const checkedRows = controllerLookup('budget').get('checkedRows');
+
+    // Do nothing with the button if more than one row is selected. (maybe change in the future)
+    if (checkedRows.length !== 1) {
+      return this.removeButton();
+    }
+
+    const category = checkedRows[0];
+    const { isMasterCategory, available, budgeted } = category.getProperties('isMasterCategory', 'available', 'budgeted');
+
+    // TODO: Test
+    console.log(budgeted);
+
+    // We can't budget for master categories
+    // We don't need this if available is not negative
+    if (isMasterCategory || available >= 0) {
+      return this.removeButton();
+    }
+
+    // Search for the button
+    let button = $(`.${BUTTON_CLASS}`);
+
+    // Create button if it doesn't exist already
+    if (!button.length) {
+      button = $('<button>', {
+        class: `budget-inspector-button ${BUTTON_CLASS}`
+      }).click(this.coverFromFuture.bind(this, category));
```

Comment:
So, in general -- I would avoid using `bind` and even more, avoid default arguments when using `bind` when possible. For this use case, I would just do:

```
.click(() => {
  this.coverFromFuture(category);
});
```

using the lambda function preserves the value of `this` for you :D